src/shogun/base/SGObject.h

```diff
@@ -278,6 +283,120 @@ class CSGObject
 	 */
 	void build_gradient_parameter_dictionary(CMap<TParameter*, CSGObject*>* dict);
 
+	/** Sets a value for a parameter which can be identified by a tag.
+	 *
+	 * @param _tag contains name and type information of a parameter
+	 * @param value value of the parameter
+	 * @return previous value if the parameter existed corresponding to the input tag,
+	 * otherwise returns current value.
+	 */
+	template <typename T>
+	inline T set(Tag<T>& _tag, const T& value)
+	{
+		return recall_type<T>(_set(_tag, erase_type(value)));
+	}
+
+	/** Gets the value of a parameter corresponding to a tag.
+	 *
+	 * @param _tag contains name and type information of a parameter
+	 * @return value of the parameter corresponding to the input tag
+	 */
+	template <typename T>
+	inline T get(Tag<T>& _tag) const
+	{
+		const Any value = _get(_tag);
+		try
+		{
+			return recall_type<T>(value);
+		}
+		catch(std::logic_error)
+		{
+			SG_SERROR("Type is not correct.\n");
+		}
+	}
+
+	/** Sets a value for a parameter which can be identified by a name and data type.
+	 *
+	 * @param name name of the parameter
+	 * @param value value of the parameter along with type information
+	 * @return previous value if the parameter existed corresponding to the input name and type,
+	 * otherwise returns current value.
+	 */
+	template <typename T>
+	inline T set(const std::string& name, const T& value)
+	{
+		BaseTag tag(name);
+		return recall_type<T>(_set(tag, erase_type(value)));
+	}
+
+	/** Gets the value of a parameter corresponding to a name and type.
+	 *
+	 * @param name name of the parameter
+	 * @param type data type information of the parameter
+	 * @return value of the parameter corresponding to the input name and type
+	 */
+	template <typename T>
+	inline T get(const std::string& name, const Type<T>& type) const
+	{
+		BaseTag tag(name);
+		const Any value = _get(tag);
+		try
+		{
+			return recall_type<T>(value);
+		}
+		catch(std::logic_error)
+		{
+			SG_SERROR("Type is not correct.\n");
+		}
+	}
+
+	/** Checks if object has a parameter identified by a name.
+	 *
+	 * @param name name of the parameter
+	 * @return true if the parameter existed with the input name
+	 */
+	inline bool has(const std::string& name) const
+	{
+		BaseTag tag(name);
+		return _has(tag);
+	}
+
+	/** Checks if a parameter with a tag already exists.
+	 *
+	 * @param tag tag of the parameter containing name and type information
+	 * @return true if the parameter existed with the input tag
+	 */
+	template <typename T>
+	inline bool has(const Tag<T>& tag) const
+	{
+		return _has(tag);
+	}
+
+	/** Checks if a type exists for a parameter identified by a name.
+	 *
+	 * @param name name of the parameter
+	 * @param type type information for the parameter
+	 * @return true if the parameter existed with the input name and type
```

Comment:
and here, will not comment on similar issue again, please sort out all of them :)